### PR TITLE
DAOS-8008 control: set dpdk log masks to error

### PR DESCRIPTION
**Pull request by tanabarr**
Undesired log messages are being printed to stdout of daos_server. These
messages originate from DPDK which sets some of the facility log masks
to NOTICE and INFO.

This PR sets masks for all DPDK facilities to ERROR in daos_admin and
daos_engine and moves the log level cli args to a common location.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/bio/bio_xstream.c [hunk 1]
```diff
@@ -98,7 +98,7 @@ bio_spdk_env_init(void)
 	 * logging by setting level to ERROR.
 	 */
 	opts.env_context = "--log-level=lib.eal:4 --log-level=lib.user1:4"
-		" --no-telemetry";
+		" --log-level=lib.pmd:4 --no-telemetry";
```
Comment (mjmac) [inline src/bio/bio_xstream.c:101]: I'm pretty sure this should just be `pmd:4`... https://github.com/DPDK/dpdk/blob/main/lib/eal/common/eal_common_log.c#L356

Comment (tanabarr) [reply src/bio/bio_xstream.c]: thanks

**Review by johannlombardi — APPROVED** [review]

**Review by mjmac — CHANGES_REQUESTED** [review]
Unfortunately, this doesn't seem to have had any effect: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6446/2/artifact/Functional%20Hardware%20Large/ior/ior_small.py/daos_logs/test_ior_small_daos_control.wolf-110.log/*view*/

```
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin [2021-08-13 14:22:20.111151] [ DPDK EAL parameters: [2021-08-13 14:22:20.111161] daos_admin [2021-08-13 14:22:20.111168] --no-shconf [2021-08-13 14:22:20.111174] -c 0x1 [2021-08-13 14:22:20.111181] --log-level=lib.eal:6 [2021-08-13 14:22:20.111188] --log-level=lib.cryptodev:5 [2021-08-13 14:22:20.111194] --log-level=user1:6 [2021-08-13 14:22:20.111201] --log-level=lib.eal:4 [2021-08-13 14:22:20.111207] --log-level=user1:4 [2021-08-13 14:22:20.111213] --log-level=pmd:4 [2021-08-13 14:22:20.111218] --no-telemetry [2021-08-13 14:22:20.111224] --base-virtaddr=0x200000000000 [2021-08-13 14:22:20.111230] --match-allocations [2021-08-13 14:22:20.111235] --file-prefix=spdk_pid162346 [2021-08-13 14:22:20.111240] ]
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.0 on NUMA node 0
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.1 on NUMA node 0
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.2 on NUMA node 0
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.3 on NUMA node 0
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.4 on NUMA node 0
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.5 on NUMA node 0
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.6 on NUMA node 0
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.7 on NUMA node 0
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.0 on NUMA node 1
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
IOAT: ioat_rawdev_create(): Unable to allocate raw device
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.0 cannot be used
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.1 on NUMA node 1
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
IOAT: ioat_rawdev_create(): Unable to allocate raw device
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.1 cannot be used
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.2 on NUMA node 1
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
IOAT: ioat_rawdev_create(): Unable to allocate raw device
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.2 cannot be used
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.3 on NUMA node 1
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
IOAT: ioat_rawdev_create(): Unable to allocate raw device
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.3 cannot be used
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.4 on NUMA node 1
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
IOAT: ioat_rawdev_create(): Unable to allocate raw device
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.4 cannot be used
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.5 on NUMA node 1
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
IOAT: ioat_rawdev_create(): Unable to allocate raw device
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.5 cannot be used
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.6 on NUMA node 1
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
IOAT: ioat_rawdev_create(): Unable to allocate raw device
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.6 cannot be used
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.7 on NUMA node 1
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
IOAT: ioat_rawdev_create(): Unable to allocate raw device
wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.7 cannot be used
```

**Comment by tanabarr** [comment]
> Unfortunately, this doesn't seem to have had any effect: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6446/2/artifact/Functional%20Hardware%20Large/ior/ior_small.py/daos_logs/test_ior_small_daos_control.wolf-110.log/*view*/
> 
> ```
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin [2021-08-13 14:22:20.111151] [ DPDK EAL parameters: [2021-08-13 14:22:20.111161] daos_admin [2021-08-13 14:22:20.111168] --no-shconf [2021-08-13 14:22:20.111174] -c 0x1 [2021-08-13 14:22:20.111181] --log-level=lib.eal:6 [2021-08-13 14:22:20.111188] --log-level=lib.cryptodev:5 [2021-08-13 14:22:20.111194] --log-level=user1:6 [2021-08-13 14:22:20.111201] --log-level=lib.eal:4 [2021-08-13 14:22:20.111207] --log-level=user1:4 [2021-08-13 14:22:20.111213] --log-level=pmd:4 [2021-08-13 14:22:20.111218] --no-telemetry [2021-08-13 14:22:20.111224] --base-virtaddr=0x200000000000 [2021-08-13 14:22:20.111230] --match-allocations [2021-08-13 14:22:20.111235] --file-prefix=spdk_pid162346 [2021-08-13 14:22:20.111240] ]
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.0 on NUMA node 0
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.1 on NUMA node 0
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.2 on NUMA node 0
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.3 on NUMA node 0
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.4 on NUMA node 0
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.5 on NUMA node 0
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.6 on NUMA node 0
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.7 on NUMA node 0
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.0 on NUMA node 1
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
> IOAT: ioat_rawdev_create(): Unable to allocate raw device
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.0 cannot be used
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.1 on NUMA node 1
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
> IOAT: ioat_rawdev_create(): Unable to allocate raw device
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.1 cannot be used
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.2 on NUMA node 1
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
> IOAT: ioat_rawdev_create(): Unable to allocate raw device
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.2 cannot be used
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.3 on NUMA node 1
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
> IOAT: ioat_rawdev_create(): Unable to allocate raw device
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.3 cannot be used
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.4 on NUMA node 1
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
> IOAT: ioat_rawdev_create(): Unable to allocate raw device
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.4 cannot be used
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.5 on NUMA node 1
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
> IOAT: ioat_rawdev_create(): Unable to allocate raw device
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.5 cannot be used
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.6 on NUMA node 1
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
> IOAT: ioat_rawdev_create(): Unable to allocate raw device
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.6 cannot be used
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.7 on NUMA node 1
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
> IOAT: ioat_rawdev_create(): Unable to allocate raw device
> wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.7 cannot be used
> ```
> 
> I don't think these are actual errors; they're just being displayed as errors because they're printed to the stderr stream, e.g. https://github.com/DPDK/dpdk/blob/main/drivers/raw/ioat/ioat_rawdev.c#L287

any ideas how we fix this? @NiuYawei ? do we need to reinstate the allow list filter when starting engines to remove the "cannot be used" messages?

**Review by johannlombardi — CHANGES_REQUESTED** [review]
Adding Niu and Sydney in case they have any suggestions.

**Comment by NiuYawei** [comment]
> > Unfortunately, this doesn't seem to have had any effect: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6446/2/artifact/Functional%20Hardware%20Large/ior/ior_small.py/daos_logs/test_ior_small_daos_control.wolf-110.log/*view*/
> > ```
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin [2021-08-13 14:22:20.111151] [ DPDK EAL parameters: [2021-08-13 14:22:20.111161] daos_admin [2021-08-13 14:22:20.111168] --no-shconf [2021-08-13 14:22:20.111174] -c 0x1 [2021-08-13 14:22:20.111181] --log-level=lib.eal:6 [2021-08-13 14:22:20.111188] --log-level=lib.cryptodev:5 [2021-08-13 14:22:20.111194] --log-level=user1:6 [2021-08-13 14:22:20.111201] --log-level=lib.eal:4 [2021-08-13 14:22:20.111207] --log-level=user1:4 [2021-08-13 14:22:20.111213] --log-level=pmd:4 [2021-08-13 14:22:20.111218] --no-telemetry [2021-08-13 14:22:20.111224] --base-virtaddr=0x200000000000 [2021-08-13 14:22:20.111230] --match-allocations [2021-08-13 14:22:20.111235] --file-prefix=spdk_pid162346 [2021-08-13 14:22:20.111240] ]
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.0 on NUMA node 0
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.1 on NUMA node 0
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.2 on NUMA node 0
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.3 on NUMA node 0
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.4 on NUMA node 0
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.5 on NUMA node 0
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.6 on NUMA node 0
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:00:04.7 on NUMA node 0
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.0 on NUMA node 1
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
> > IOAT: ioat_rawdev_create(): Unable to allocate raw device
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.0 cannot be used
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.1 on NUMA node 1
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
> > IOAT: ioat_rawdev_create(): Unable to allocate raw device
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.1 cannot be used
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.2 on NUMA node 1
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
> > IOAT: ioat_rawdev_create(): Unable to allocate raw device
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.2 cannot be used
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.3 on NUMA node 1
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
> > IOAT: ioat_rawdev_create(): Unable to allocate raw device
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.3 cannot be used
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.4 on NUMA node 1
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
> > IOAT: ioat_rawdev_create(): Unable to allocate raw device
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.4 cannot be used
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.5 on NUMA node 1
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
> > IOAT: ioat_rawdev_create(): Unable to allocate raw device
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.5 cannot be used
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.6 on NUMA node 1
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
> > IOAT: ioat_rawdev_create(): Unable to allocate raw device
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.6 cannot be used
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin IOAT: ioat_rawdev_probe(): Init 0000:80:04.7 on NUMA node 1
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin rte_rawdev_pmd_allocate(): Unable to allocate memory for rawdev
> > IOAT: ioat_rawdev_create(): Unable to allocate raw device
> > wolf-110.wolf.hpdd.intel.com ERROR 2021/08/13 14:22:20 /usr/bin/daos_admin EAL: Requested device 0000:80:04.7 cannot be used
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I don't think these are actual errors; they're just being displayed as errors because they're printed to the stderr stream, e.g. https://github.com/DPDK/dpdk/blob/main/drivers/raw/ioat/ioat_rawdev.c#L287
> 
> any ideas how we fix this? @NiuYawei ? do we need to reinstate the allow list filter when starting engines to remove the "cannot be used" messages?
> also why are they being printed when we set pmd log mask to 4 (ERR)?

I don't see why the messages are printed even when pmd log mask is set to ERR. I think reinstating the allowed list to avoid scanning unrelated devices could be a safer way.

**Comment by tanabarr** [comment]
logging output will be greatly improved by landing this and https://github.com/daos-stack/dpdk/pull/20

**Review by NiuYawei — APPROVED** [review]

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**